src/main/java/com/clouway/asynctaskscheduler/gae/RoutingEventDispatcher.java

```diff
@@ -87,8 +111,10 @@ public void dispatchAsyncEvent(String eventClassAsString, String eventAsJson) th
 
   private void dispatchListeners(AsyncEvent<AsyncEventHandler> event) {
     List<? extends AsyncEventListener> listeners  = listenersFactory.create(event.getClass());
+    int id = 0;
     for (AsyncEventListener listener : listeners) {
-      listener.onEvent(event);
+      taskScheduler.get().add(AsyncTaskOptions.event(event).eventListenerId(id)).now();
```

Comment:
i'm thinking of a validation that all listeners are added once even if something fails while adding listener to taskQueue